plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/N4JSExpressionValidator.xtend

```diff
@@ -1225,6 +1225,9 @@ class N4JSExpressionValidator extends AbstractN4JSDeclarativeValidator {
 		// avoid consequential errors
 		if (S === null || T === null || T instanceof UnknownTypeRef || S instanceof UnknownTypeRef) return;
 
+		// casts from dynamically typed variables introduce more type information. Hence do not show warning.
+		if (S.isDynamic) return;
```

Comment:
Maybe we could say `if (S.isDynamic XOR T.isDynamic) ...`, here, in order to avoid a warning also in cases like this:
```javascript
let z: string;
(z as any+);
```

Reply:
good idea!